R/merge_sce_list.R

```diff
@@ -0,0 +1,138 @@
+#' Merge a list of SCEs as preparation for formal integration
+#'
+#' This function takes a named (ideally by a form of library ID) list of
+#'  SingleCellExperiment (SCE) objects and merges them into one SCE object.
+#'  All names in the colData slot must match across provided SCE objects.
+#'  The resulting colData slot will contain information in the given batch column
+#'  to differentiate originating SCE objects. Only features (genes) that are present
+#'  in all provided SCE objects will be retained. To preserve original feature-level
+#'  information, all final rowData slot column names will be appended with the
+#'  given SCE's name, as `<column_name>-<sce_name>`.
+#'
+#' @param sce_list A named list of SingleCellExperiment objects
+#' @param batch_column A character value giving the resulting colData column name
+#'  to differentiate originating SingleCellExperiment objects Default value is `batch`.
+#' @param retain_coldata_columns_main A vector of colData columns which should be retained
+#'  in the main experiment of the final merged SCE object.
+#' @param retain_coldata_columns_altexps A vector of colData columns which should be retained
+#'  in the altExps experiments of the final merged SCE object.
+#' @param retain_altexps A logical indicating whether any present altExp slots should
+#'   be retained in the merged SCE. Default is `TRUE`.
+#' @param preserve_rowdata_columns A vector of column names that appear in originating
+#'  SingleCellExperiment objects' rowData slots which should not be renamed with
+#'  the given SingleCellExperiment object name. These are generally columns which are
+#'  not specific to the given library's preparation or statistics. For example,
+#'  such an array might contain items like "Gene", "ensembl_ids", "gene_symbol", etc.
+#'
+#' @return A SingleCellExperiment object containing all SingleCellExperiment objects
+#'   present in the inputted list
+#' @export
+#'
+#' @import SingleCellExperiment
+merge_sce_list <- function(sce_list = list(),
+                           batch_column = "batch",
+                           retain_altexps = TRUE,
+                           retain_coldata_columns_main = c("sum",
+                                                           "detected",
+                                                           "total",
+                                                           "subsets_mito_sum",
+                                                           "subsets_mito_detected",
+                                                           "subsets_mito_percent",
```

Comment:
```suggestion
                                                           "subsets_mito_percent"
                                                           "prob_compromised",
```
I think we probably also want to keep the probability compromised values determined from miQC.